azext_iot/monitor/handlers/common_handler.py

```diff
@@ -70,3 +73,24 @@ def _should_process_interface(self, interface_name):
 
         # only process if the expected and actual interface name match
         return expected_interface_name == interface_name
+
+    def _should_process_module(self, module_id):
+        expected_module_id = self._common_handler_args.module_id
+
+        # if no filter is specified, then process all modules
+        if not expected_module_id:
+            return True
+
+        if expected_module_id and expected_module_id != module_id:
+            if "*" in expected_module_id or "?" in expected_module_id:
+                regex = (
+                    re.escape(expected_module_id)
+                    .replace("\\*", ".*")
+                    .replace("\\?", ".")
+                    + "$"
+                )
+                if not re.match(regex, module_id):
+                    return False
+            else:
+                return False
```

Comment:
Lets pull 85 - 96 into its own function because it looks copy-pasted from `should_process_device`

Reply:
infact, we could extract everything below 81 out into a function